File: JavaScript Implementation/symbol-table/Symbol.js
class Symbol {
    constructor(name, type=null) {
        this.name = name;
        this.type = type;
    }

    getName() {
        return this.name;
    }

    getType() {
        return this.type;
    }

    toString() {
        return `Symbol(${this.getName()}, ${this.getType()})`;
    }

    static create(name, type) {
        return new this(name, type);
    }
}

module.exports = Symbol;

File: Java Implementation/src/symbol_table/Symbol.java
package symbol_table;

public class Symbol {
    private String name;
    private Object type;

    public Symbol(String name, Object type) {
        this.name = name;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public Object getType() {
        return type;
    }
}
